ui/js/search.js

```diff
@@ -100,25 +183,48 @@ require(['jquery','oae.core', 'jquery.history'], function($, oae) {
      * This will only be executed when the page is loaded.
      */
     var initSearch = function() {
-        // We parse the URL fragment that's inside of the current History.js state.
-        // The expected URL structure is `/search/<query>?types=type1,type2`
-        var url = History.getState().cleanUrl;
-        var initialState = $.url(url);
-        var query = initialState.segment().slice(1).join('/');
-        var types = (initialState.param().types || '').split(',');
-        // Replace the current History.js state to have the query and type refinement data. This
-        // is necessary because a newly loaded page will not contain the data object in its
-        // state. Calling the replaceState function will automatically trigger the statechange
-        // event, which will take care of the actual search. However, we also need to add a random
-        // number to the data object to make sure that the statechange event is triggered after
-        // a page reload.
-        History.replaceState({
-            'query': query,
-            'types': types,
-            '_': Math.random()
-        }, $('title').text(), url);
-
         oae.api.util.template().render($('#search-list-header-template'), null, $('#search-list-header'));
+        oae.api.util.template().render($('#search-lhnavigation-template'), null, $('#search-lhnavigation'));
+
+        // We should intelligently detect that if someone goes to /search/<search query>, we should
+        // automatically send them to /search/all?q=<search query>
+        var state = History.getState()
+        var url = $.url(History.getState().cleanUrl);
+        var path = url.attr('path').split('/');
+        var queryString = url.attr('query');
+
+        /*!
+         * This is backward compatibility handling for links that are sent to OAE. Previously, a
+         * link could be sent to /search/<query> to send a user to the search page with a pre-
+         * loaded query. Now those search endpoints look like /search/all/<query> and
+         * /search/my/<query>.
+         *
+         * To transition to this, we look if the query looks like /search/<query> where <query> is
+         * not "all" or "my". If so, we rewrite the URL to /search/all/<query>.
```

Comment:
Can you add a deprecated notice here and a TODO for removing this in the next release? Can you also send a note to this list about this backwards incompatibility and when it will be removed?